### PR TITLE
adding utils for working with cookies

### PR DESCRIPTION
**Pull request by aintgoin2goa**
@bjfletcher @phamann

### CODE REVIEW

File: utils/js/cookies.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+function getOptions (options){
+	return Object.keys(options).map(key => {
+		let newKey = key;
+		if(key === 'maxAge'){
+			newKey = 'max-age';
+		}
+
+		if(options[key] instanceof Date){
+			options[key] = options[key] = options[key].toUTCString();
```
Comment (debugwand) [inline utils/js/cookies.js]: `options[key] = options[key] = options[key].toUTCString();`  - why 2`options[key] = `

File: utils/js/cookies.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		}
+
+		return `${newKey}=${options[key]}`;
+	});
+}
+
+function set (name, value, opts){
+	const options = getOptions(opts).join('; ');
+	const cookie = `${encodeURIComponent(name)}=${encodeURIComponent(value)}; ${options}`;
```
Comment (aintgoin2goa) [inline utils/js/cookies.js]: Cos cookies are supposed to be url encoded according to the spec.  Browsers handle it either way in most cases, but just trying to be correct

File: utils/js/cookies.js (hunk 1 above)
Comment (aintgoin2goa) [inline utils/js/cookies.js]: Cos I'm an idiot who types things twice types things twice

File: utils/js/cookies.js (hunk 2 above)
Comment (ironsidevsquincy) [inline utils/js/cookies.js]: Then we should un-encode them when `get`ting

Comment (ironsidevsquincy) [inline utils/js/cookies.js]: And presumably also encode the name before `get`ting and `has`ing

Comment (aintgoin2goa) [inline utils/js/cookies.js]: yeah good point

**Comment by bjfletcher** [comment]
Great, thanks Paul!

Can we add a comment at the top of the utility:

https://github.com/WICG/async-cookies-api

that we can use as reference as potentially something we may want to use in the future.

(Note that we can't use that polyfill just yet because it uses `async`.)

**Pull request merged**